File: chapter-02-simple_md/cpp-simpleMD/ex2-cutoff/plot_results.m
clear; close all;
load 7A/thermo.out; energy_7A = sum(thermo(2:end,2:3),2);
load 9A/thermo.out; energy_9A = sum(thermo(2:end,2:3),2);
load 12A/thermo.out; energy_12A = sum(thermo(2:end,2:3),2);
load 15A/thermo.out; energy_15A = sum(thermo(2:end,2:3),2);

fluctuation_7A = (energy_7A - mean(energy_7A))/abs(mean(energy_7A));
fluctuation_9A = (energy_9A - mean(energy_9A))/abs(mean(energy_9A));
fluctuation_12A = (energy_12A - mean(energy_12A))/abs(mean(energy_12A));
fluctuation_15A = (energy_15A - mean(energy_15A))/abs(mean(energy_12A));

x = ["7 A", "9 A", "12 A","15 A"];
y = [std(fluctuation_7A),std(fluctuation_9A),std(fluctuation_12A),std(fluctuation_15A)];

figure;
bar(x,y); 
ylabel('总能量相对涨落');
set(gca,'fontsize',10)

File: chapter-02-simple_md/cpp-simpleMD/ex2-cutoff/plot_results.py
import numpy as np
import matplotlib.pyplot as plt

data=[]
data.append(np.loadtxt('7A/thermo.out'))
data.append(np.loadtxt('9A/thermo.out'))
data.append(np.loadtxt('12A/thermo.out'))
data.append(np.loadtxt('15A/thermo.out'))

fluctuation_value = np.zeros(4)
for n in range(4):
    energy = np.sum(data[n][1:,1:3], axis = 1)
    fluctuation_value[n] = np.std((energy - np.mean(energy))/np.abs(energy))
    print(fluctuation_value[n])

plt.figure(figsize=(8, 6))

plt.semilogy(np.array((7,9,12,15)), fluctuation_value, '-o', linewidth=2)
plt.xlabel('Cutoff radius ($\mathrm{\AA}$)', fontsize=15)
plt.ylabel('Magnitude of relative energy fluctuations', fontsize=15)

plt.tight_layout()
plt.savefig('fig-c02-cutoff.pdf')
